rasa_core/policies/embedding_policy.py

```diff
@@ -0,0 +1,848 @@
+        # type: (List[DialogueStateTracker], Domain, **Any) -> None
+        """Continues training an already trained policy."""
+
+        batch_size = kwargs.get('batch_size', 5)
+        epochs = kwargs.get('epochs', 50)
+
+        num_samples = batch_size - 1
+        num_prev_examples = len(training_trackers) - 1
+        for _ in range(epochs):
+            sampled_idx = np.random.choice(range(num_prev_examples),
+                                           replace=False,
+                                           size=min(num_samples,
+                                                    num_prev_examples))
+            trackers = [training_trackers[i]
+                        for i in sampled_idx] + training_trackers[-1:]
+            training_data = self.featurize_for_training(trackers,
+                                                        domain)
+            (batch_a, batch_pos_b, batch_c, actions_for_b
+             ) = self._prepare_data_for_training(training_data)
+
+            batch_b = self._create_batch_b(batch_pos_b, actions_for_b)
+
+            # fit to one extra example using updated trackers
+            self.session.run(self.train_op,
+                             feed_dict={self.a_in: batch_a,
+                                        self.b_in: batch_b,
+                                        self.c_in: batch_c,
+                                        self.is_training: True})
+
+    def predict_action_probabilities(self, tracker, domain):
+        # type: (DialogueStateTracker, Domain) -> List[float]
+        """Predicts the next action the bot should take
+        after seeing the tracker.
+
+        Returns the list of probabilities for the next actions"""
+        if self.session is None:
+            logger.error("There is no trained tf.session: "
+                         "component is either not trained or "
+                         "didn't receive enough training data")
+            return []
+
+        data_X = self.featurizer.create_X([tracker], domain)
+
+        X, extras = self._create_X_extras(data_X)
+        all_Y_d = self._create_all_Y_d(X.shape[1])
+        all_Y_d_x = np.stack([all_Y_d for _ in range(X.shape[0])])
+
+        _sim = self.session.run(
+            self.sim_op, feed_dict={self.a_in: X,
+                                    self.b_in: all_Y_d_x,
+                                    self.c_in: extras}
+        )
+        return _sim[0, -1, :].tolist()
+
+    def persist(self, path):
+        # type: (Text) -> None
+        """Persists the policy to a storage."""
+        super(EmbeddingPolicy, self).persist(path)
+
+        if self.session is None:
+            warnings.warn("Persist called without a trained model present. "
+                          "Nothing to persist then!")
+            return
+
+        file_name = 'tensorflow_embedding.ckpt'
+        checkpoint = os.path.join(path, file_name)
+
+        try:
+            os.makedirs(os.path.dirname(checkpoint))
```

Comment:
if seen this in the code base before, but given that the try catch is quite large, I'd vote for:
```
if not os.path.exists(path):
  os.makedirs(path)
```

Reply:
it is the code from https://github.com/RasaHQ/rasa_core/blob/811d42b1d29c40a42d7dee345572ce73330b8360/rasa_core/utils.py#L133
I will call this function here

Reply:
even better :+1: